src/ess/sans/i_of_q.py

```diff
@@ -222,6 +226,29 @@ def merge_spectra(
     return CleanSummedQ[ScatteringRunType, IofQPart](out.squeeze())
 
 
+def dummy_merge_runs(
+    data: CleanSummedQ[ScatteringRunType, IofQPart]
+) -> FinalSummedQ[ScatteringRunType, IofQPart]:
+    return FinalSummedQ[ScatteringRunType, IofQPart](data)
+
+
+def _merge_events_or_histograms(a: sc.DataArray, b: sc.DataArray) -> sc.DataArray:
+    if a.bins is not None:
+        return a.bins.concatenate(b)
+    return a + b
+
+
+def merge_multiple_runs(
+    data: sciline.Series[
+        Filename[ScatteringRunType], CleanSummedQ[ScatteringRunType, IofQPart]
+    ],
+) -> FinalSummedQ[ScatteringRunType, IofQPart]:
+    """ """
+    return FinalSummedQ[ScatteringRunType, IofQPart](
+        reduce(_merge_events_or_histograms, data.values())
+    )
```

Comment:
Consider using https://scipp.github.io/generated/functions/scipp.reduce.html.

Reply:
I tried something like
```Py
reducer = sc.reduce(data.values())
out = reducer.bins.concat() if reducer.bins is not None else reducer.sum()
```
but `reducer.bins` is never `None`, even if the elements inside `data.values()` are dense data.

So I went back to `functools.reduce`.

Reply:
Why can't use use it as `if data.values()[0].bins is None`?

Reply:
To be clear: Doing it in the way you chose is *extremely* inefficient if you have many files (please give it a try!).

Reply:
Actually, I have to qualify my statement: When summing, I think the current implementation of `sc.reduce` is actually at a disadvantage, compared to `functools.reduce`, but for concatenation it is $N^2$.